flashinfer/fmha_v2.py

```
@@ -0,0 +1,551 @@
+        v_head_dim: int,
+        out: Optional[torch.Tensor],
+        lse: Optional[torch.Tensor],
+        return_lse: bool,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        # FMHAv2 softmax stats: (max, sum_exp) per token/head, ragged format.
+        lse_shape = (q.size(0), q.size(1), 2)
+        if lse is not None:
+            check_shape_dtype_device(lse, lse_shape, torch.float32, q.device, "lse")
+        elif return_lse:
+            lse = torch.empty(lse_shape, dtype=torch.float32, device=q.device)
+        out_shape = q.shape[:-1] + (v_head_dim,)
```

Ruff (0.15.21)

[warning] Consider `(*q.shape[:-1], v_head_dim)` instead of concatenation

Replace with `(*q.shape[:-1], v_head_dim)`

(RUF005)